File: Python/Produto.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    
    def __add__(self, outro):
        if isinstance(outro, Produto):
            return Produto(f"{self.nome} + {outro.nome}", self.preco + outro.preco)
        return NotImplemented

    def __str__(self):
        return f"Produto: {self.nome}, Preço: {self.preco:.2f}"


produto1 = Produto("Produto A", 42.50)
produto2 = Produto("Produto B", 56.50)

produto_soma = produto1 + produto2
print(produto_soma)

File: Go/Produto.go
package listaRicardo

import (
	"fmt"
)

type Produto struct {
	nome  string
	preco float64
}

func somar(produto1, produto2 Produto) Produto {
	return Produto{
		nome:  produto1.nome + " + " + produto2.nome,
		preco: produto1.preco + produto2.preco,
	}
}

func (p Produto) String() string {
	return fmt.Sprintf("Produto: %s, Preço: %.2f", p.nome, p.preco)
}

func main() {
	produto1 := Produto{"Produto A", 50.50}
	produto2 := Produto{"Produto B", 49.50}

	produtoSoma := somar(produto1, produto2)
	fmt.Println(produtoSoma.String())
}
